### PR TITLE
Add new_test/test_atomic_hint.F90

### PR DESCRIPTION
**Pull request by seyonglee**
- NVHPC 22.11:
            - C test failed: line 25: error: invalid text in pragma
            - Fortran test failed: NVFORTRAN-S-0034-Syntax error at or near identifier hint
        - LLVM 15.0.0: C test passed.
        - LLVM 17.0.0: C test passed.
        - GCC 12.2.1:
            - Both C and Fortran tests passed.
        - XL 16.1.1-10:
            - C test passed, but test ran with one thread, so the results are not conclusive
            - Fortran test failed:  line 37.13: 1515-019 (S) Syntax is incorrect.

### CODE REVIEW

File: tests/5.0/atomic/test_atomic_hint.F90 [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    !$omp parallel num_threads(OMPVV_NUM_THREADS_HOST) default(shared)
+       IF (omp_get_thread_num() .EQ. 0) THEN
+          num_threads = omp_get_num_threads()
+       END IF
+       !$omp atomic hint(omp_sync_hint_contended+omp_sync_hint_nonspeculative)
+          x = x + 1
+       !$omp end atomic 
+    !$omp end parallel
+
+    OMPVV_ERROR_IF(num_threads .LT. 0, "Test ran with invalid number of teams (less than zero)")
```
Comment (spophale) [inline tests/5.0/atomic/test_atomic_hint.F90:68]: threads (not teams)

File: tests/5.0/atomic/test_atomic_hint.F90 [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    !$omp parallel num_threads(2) default(shared)
+       IF (omp_get_thread_num() .EQ. 0) THEN
+          num_threads = omp_get_num_threads()
+       END IF
+       !$omp atomic hint(omp_sync_hint_uncontended)
+          x = x + 1
+       !$omp end atomic 
+    !$omp end parallel
+
+    OMPVV_ERROR_IF(num_threads .LT. 0, "Test ran with invalid number of teams (less than zero)")
```
Comment (spophale) [inline tests/5.0/atomic/test_atomic_hint.F90:42]: threads (not teams)

**Review by spophale — CHANGES_REQUESTED** [review]
Messages need to be corrected.

File: tests/5.0/atomic/test_atomic_hint.F90 [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+             a(2) = a(2) + 1
+          !$omp end atomic 
+       END IF
+       !$omp atomic hint(omp_sync_hint_speculative)
+          a(i) = a(i) + i
+       !$omp end atomic 
+    END DO
+    !$omp end parallel do
+
+    OMPVV_ERROR_IF(num_threads .LT. 0, "Test ran with invalid number of teams (less than zero)")
```
[the remark this answers is not in the document]
Comment (seyonglee) [reply tests/5.0/atomic/test_atomic_hint.F90]: Updated; thanks.

File: tests/5.0/atomic/test_atomic_hint.F90 (hunk 1 above)
Comment (seyonglee) [reply tests/5.0/atomic/test_atomic_hint.F90]: Updated; thanks.

File: tests/5.0/atomic/test_atomic_hint.F90 (hunk 2 above)
Comment (seyonglee) [reply tests/5.0/atomic/test_atomic_hint.F90]: Updated; thanks.

**Comment by seyonglee** [comment]
> Messages need to be corrected.

Corrected

**Review by spophale — APPROVED** [review]

**Review by mjcarr458 — APPROVED** [review]

**Pull request merged**